safe/load.go

```diff
@@ -15,19 +17,45 @@ func Load(password []byte, backend backends.Client, encryptionClient crypto.Clie
 		password: password,
 	}
 
-	ciphertext, err := s.backend.Load()
-
-	if _, ok := err.(*errors.SafeNotFound); ok {
-		s.Accounts = make(map[string]Account)
-		return &s, nil
-	}
+	data, err := s.backend.Load()
 	if err != nil {
+		if _, ok := err.(*errors.SafeNotFound); ok {
+			s.Accounts = make(map[string]Account)
+			return &s, nil
+		}
 		return nil, err
 	}
 
-	plaintext, err := s.crypto.Decrypt(ciphertext, password)
+	safeFile := safeFile{}
+	if err := json.Unmarshal(data, &safeFile); err == nil {
+		// Unmarshalling did succeed -> Safe uses new format
+		// Do nothing
+	} else {
+		safeFile.Ciphertext = data
+		// If we don't have a config, use OpenPGP for backwards-compatibility
+		defaultOpenPGPConfig := crypto.DefaultConfigWithType("aes_openpgp")
+		safeFile.Config = &defaultOpenPGPConfig
+	}
+
+	upgradeNeeded := false
+	plaintext, err := s.crypto.Decrypt(safeFile.Ciphertext, password)
```

Comment:
This will fail in the case where a safe is created with no config available, and sometime later a config is added and the encryption type changed.

Repo
1. Remove any existing `pick.safe` and `config.toml`
2. `add` an account
3. Add a config with `Make config`
4. Change `encryption.type` in the config so `aes-196|aes-128`
5. `cat` the account
6. Notice the safe is upgraded
7. `cat` the account again

    > Unable to unlock safe

At this point the safe will be unreadable, even if the config settings are reverted. I'm not opposed to punting on this for now, as I'd imagine this is a rare use case, but if we go that route it must be documented clearly.